### ISSUE TITLE
Documentation: installation of dependencies when installing from source.

### ISSUE THREAD

**Issue by jmp75** [zricethezav/gitleaks#1029]
**Describe the solution you'd like**

Additional guidance in the documentation on how to install go dependencies to build and install gitleak.

**Additional context**

I am working on a Debian OS. I assume Ubuntu would be a similar experience.

I can figure out I need `sudo apt golang` to then try `make build`, fair enough. But installing the following packages is less obvious to folks not familiar with the  `go` ecosystem:

```text
config/config.go:4:2: package embed is not in GOROOT (/usr/lib/go-1.15/src/embed)
detect/detect.go:8:2: package io/fs is not in GOROOT (/usr/lib/go-1.15/src/io/fs)
```

I cannot find where to source these from Debian packages (admittedly a Debian package usability issue as well), and figuring out how to use `go get` is a bit of unnecessary friction.

If there were guidelines to ease installation from source, this would be good.

cc @zricethezav

**Comment by weineran** [comment]
@jmp75 Agreed the instructions on installation from source are a bit sparse.

For your case, try upgrading your go version to 1.19 first. This is supposed to be enforced by the go.mod file here: https://github.com/zricethezav/gitleaks/blob/e93d8cb6967f314bed1b5e2904234f6aa25a3510/go.mod#L3
But it doesn't give the meaningful error that I'd hope for, at least in some cases.

As you are slogging through this, if you are motivated to open a PR with more thorough instructions we would love that. Possibly merits a separate .md file like BUILD.md, which we could link to from the main README.md

**Comment by ericfrederich** [comment]
For what it's worth, I think the preferred method of installing on Ubuntu is to download it from golangs website and follow their simple instructions which is not much more than a tar command and setting an environment variable.

I think when you install it from apt you're getting some attempt to integrate golangs with gcc or something, but I could be wrong.

**Comment by jmp75** [comment]
Thank you both for this information, I'll see if I can contribute. I've also notified a few colleagues in my IT and security department about gitleak, so some contributions may come from them too. 

As an aside I wrote a small [blog post](https://jmp75.github.io/work-blog/posts/20221111-gitleaks-config/) on my initial steps with gitleak.
